e2e/suite_test.go

```diff
@@ -317,16 +327,123 @@ var _ = Describe("pvc-autoresizer", func() {
 		By("checking the disk resizing")
 		checkDiskResize(pvcName, "2Gi", true)
 	})
+
+	It("should mutate the PVC size based on the same initial-resize-group-by PVC spec in the same namespace", func() {
+		// small size PVC
+		pvcName := "resize-group-pvc1"
+		sc := "topolvm-provisioner-annotated"
+		mode := string(corev1.PersistentVolumeFilesystem)
+		request := "2Gi"
+		limit := "10Gi"
+		threshold := "50%"
+		increase := "1Gi"
+		storageLimit := ""
+		initialResizeGroupByAnnotation := "test-group"
+		groupXLabel := map[string]string{
+			initialResizeGroupByAnnotation: "group-x",
+		}
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		// large size PVC
+		pvcName = "resize-group-pvc2"
+		request = "3Gi"
+		limit = "10Gi"
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		// large size PVC but other namespace
+		pvcName = "resize-group-pvc3"
+		request = "4Gi"
+		limit = "10Gi"
+		resources2 = createPodPVC2(resources2, pvcName, sc, mode, pvcName,
+			request, limit, threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		// large size PVC but other group
+		pvcName = "resize-group-pvc4"
+		request = "4Gi"
+		limit = "10Gi"
+		groupYLabel := map[string]string{
+			initialResizeGroupByAnnotation: "group-y",
+		}
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupYLabel)
+
+		// Newly created small PVC
+		pvcName = "resize-group-pvc5"
+		request = "1Gi"
+		limit = "10Gi"
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		By("checking the PVC size is mutated")
+		checkDiskResize(pvcName, "3Gi", true)
+
```

Comment:
I should add a test case to check that the resize is limited by the storage limit annotation.